File: ゼミ用/05製造/api/src/graph-dev/request/EditAccountRequest.php
<?php

/**
 * ユーザ編集処理の値を保持する
 */
class EditAccountRequest
{
    private $userId;
    private $userName;
    private $password;
    private $version;
    private $errorMsg = array();

    /**
     * ユーザ編集処理の値を格納します
     */
    public function __construct()
    {
    }

    public function getUserId()
    {
        return $this->userId;
    }

    public function setUserId($userId)
    {
        $this->userId = $userId;
    }

    public function getUserName()
    {
        return $this->userName;
    }

    public function setUserName($userName)
    {
        $this->userName = $userName;
    }

    public function getPassword()
    {
        return $this->password;
    }

    public function setPassword($password)
    {
        $this->password = $password;
    }

    public function getVersion()
    {
        return $this->version;
    }

    public function setVersion($version)
    {
        $this->version = $version;
    }

    public function getErrorMsg()
    {
        return $this->errorMsg;
    }

    /**
     * バリデーションチェックをします
     */
    public function validation()
    {
        $validationFlg = false;

        try {
            // 入力項目チェック
            if (empty($this->userId)) {
                array_push($this->errorMsg, 'エラー ：必須項目が入力されていません');
                $validationFlg = true;
            }

            // ユーザIDの文字数チェック
            if (strlen($this->userId) != Constant::USER_ID_DIGIT + strlen(Constant::USER_ID_STR)) {
                array_push($this->errorMsg, 'エラー ：ユーザIDは' . Constant::USER_ID_DIGIT + strlen(Constant::USER_ID_STR) . '文字で入力してください');
                $validationFlg = true;
            }

            // ユーザ名の文字数チェック
            if (100 < strlen($this->userName)) {
                array_push($this->errorMsg, 'エラー ：ユーザ名は100文字以内で入力してください');
                $validationFlg = true;
            }

            // パスワードの文字数チェック
            if (20 < strlen($this->password)) {
                array_push($this->errorMsg, 'エラー ：パスワードは20文字以内で入力してください');
                $validationFlg = true;
            }
        } catch (Exception $e) {
            $validationFlg = true;
        }

        return $validationFlg;
    }
}

File: ゼミ用/05製造/eeldev/eelPy/api/request/EditAccountRequest.py
from ..common.Constant import Constant


# /**
# * ユーザ編集処理の値を保持する
# */
class EditAccountRequest:
    errorMsg = []

    # /**
    # * ユーザ編集処理の値を格納します
    # */
    def __init__(self, userId, userName, password, version):
        self.userId = userId
        self.userName = userName
        self.password = password
        self.version = version

    def getUserId(self):
        return self.userId

    def getUserName(self):
        return self.userName

    def getPassword(self):
        return self.password

    def getVersion(self):
        return self.version

    def getErrorMsg(self):
        return self.errorMsg

    # /**
    # * バリデーションチェックをします
    # */
    def validation(self):
        validationFlg = False

        try:
            # 入力項目チェック
            if (len(self.userId) == 0):
                self.errorMsg.append('エラー ：必須項目が入力されていません')
                validationFlg = True

            # ユーザIDの文字数チェック
            if (Constant.USER_ID_DIGIT + len(Constant.USER_ID_STR) < len(self.userId)):
                self.errorMsg.append(
                    'エラー ：ユーザIDは' + Constant.USER_ID_DIGIT + len(Constant.USER_ID_STR) + '文字以内で入力してください')
                validationFlg = True

            # ユーザ名の文字数チェック
            if (100 < len(self.userName)):
                self.errorMsg.append('エラー ：ユーザ名は100文字以内で入力してください')
                validationFlg = True

            # パスワードの文字数チェック
            if (20 < len(self.password)):
                self.errorMsg.append('エラー ：パスワードは20文字以内で入力してください')
                validationFlg = True            

        except Exception as e:
            validationFlg = True

        return validationFlg
